### PR TITLE
add option/cache to schema

### PR DESCRIPTION
**Pull request by gadfort**
adds control of cache location, this is needed for the server to operate properly.

### CODE REVIEW

File: siliconcompiler/package.py [hunk 1]
```diff
@@ -44,7 +45,13 @@ def path(chip, package, quiet=True):
         return path
 
     # location of the python package
-    cache_path = os.path.join(Path.home(), '.sc', 'cache')
+    cache_path = chip.get('option', 'cache')
+    if cache_path:
+        cache_path = chip.find_files('option', 'cache', missing_ok=True)
+        if not cache_path:
```
Comment (troibe) [inline siliconcompiler/package.py:51]: So can cache path can be both a relative and absolute path?

File: siliconcompiler/schema/schema_cfg.py [hunk 2]
```diff
@@ -2187,6 +2187,20 @@ def schema_option(cfg):
...
+            sctype='file',
+            scope='job',
+            shorthelp="User cache directory",
+            switch="-cache <file>",
+            example=[
+                "cli: -cache /home/user/.sc/cache",
+                "api: chip.set('option', 'cache', '/home/user/.sc/cache')"],
+            schelp="""
+            Filepath to cache used for package data sources. If the
+            credentials parameter is empty, the remote processing client program
```
Comment (troibe) [inline siliconcompiler/schema/schema_cfg.py:2200]: The credentials part might be a copy paste error?

File: tests/package/test_package.py [hunk 3]
```diff
@@ -3,10 +3,15 @@
...
+import os
 
 
-def cache_path(path, ref, chip=None, quiet=True):
+def cache_path(path, ref, chip=None, quiet=True, cache=None):
     chip = chip or siliconcompiler.Chip('test')
+    chip.set('option', 'cache', cache)
+
+    if not cache:
+        cache = Path.home().joinpath('.sc/cache')
```
Comment (troibe) [inline tests/package/test_package.py:14]: Can't hurt to use `default_cache_dir()` here as well.

**Review by troibe — APPROVED** [review]
Looks good. Two minor minor suggestions.

File: siliconcompiler/package.py (hunk 1 above)
Comment (gadfort) [reply siliconcompiler/package.py:51]: yes don't see why not.

**Pull request merged**